diff --git a/qtui/src/CMakeLists.txt b/qtui/src/CMakeLists.txt
index 2237a3d4c..0f328ef69 100644
--- a/qtui/src/CMakeLists.txt
+++ b/qtui/src/CMakeLists.txt
@@ -43,8 +43,7 @@ IF(LIBGVC_FOUND)
     # Note, however, that the graphviz plugins often live in a
     # subdirectory graphviz/ beneath the main library path.
     SET(LINK_LIBS ${LINK_LIBS} ${LIBGVC_LIBRARIES})
-    SET(LINK_LIBS ${LINK_LIBS} gvplugin_core gvplugin_dot_layout gvplugin_neato_layout)
-    LINK_DIRECTORIES("${LIBGVC_LIBDIR}" "${LIBGVC_LIBDIR}/graphviz")
+    LINK_DIRECTORIES("${LIBGVC_LIBDIR}")
 
     # Ubuntu focal (20.04 LTS) has an error where the graphviz plugins
     # fail to link due to missing __XXX_finite symbols from the math
diff --git a/qtui/src/graphviz.cpp b/qtui/src/graphviz.cpp
index 7a168ed4b..36635db7e 100644
--- a/qtui/src/graphviz.cpp
+++ b/qtui/src/graphviz.cpp
@@ -40,7 +40,7 @@
 // Define LIBGVC_DYNAMIC_PLUGINS if you wish to load plugins dynamically.
 // This requires (amongst other things) the presence of the file config6,
 // which lists all available plugins.
-// #define LIBGVC_DYNAMIC_PLUGINS 1
+#define LIBGVC_DYNAMIC_PLUGINS 1
 
 #ifndef LIBGVC_DYNAMIC_PLUGINS
 #if defined(_WIN32)
