diff --git a/qtui/src/packets/facetgraphtab.cpp b/qtui/src/packets/facetgraphtab.cpp
index d2ba74340..a5f2b8c77 100644
--- a/qtui/src/packets/facetgraphtab.cpp
+++ b/qtui/src/packets/facetgraphtab.cpp
@@ -294,7 +294,7 @@ void FacetGraphTab::refresh() {
     }
 
     char* svg;
-    unsigned svgLen;
+    size_t svgLen;
 
 #ifdef LIBGVC_DYNAMIC_PLUGINS
     GVC_t* gvc = gvContext();
diff --git a/qtui/src/packets/linkgraph.cpp b/qtui/src/packets/linkgraph.cpp
index b6fb1b99a..7ddd6f22f 100644
--- a/qtui/src/packets/linkgraph.cpp
+++ b/qtui/src/packets/linkgraph.cpp
@@ -223,7 +223,7 @@ void LinkGraphUI::refresh() {
     }
 
     char* svg;
-    unsigned svgLen;
+    size_t svgLen;
 
 #ifdef LIBGVC_DYNAMIC_PLUGINS
     GVC_t* gvc = gvContext();
