--- suse_macros.in.orig	2021-04-13 10:37:40.045082922 +0000
+++ suse_macros.in	2021-04-13 10:37:54.828940956 +0000
@@ -12,8 +12,8 @@
 %_firmwaredir           /lib/firmware
 
 # package build macros
-%make_install           make install DESTDIR=%{?buildroot}
-%makeinstall            make DESTDIR=%{?buildroot:%{buildroot}} install
+%make_install           make install DESTDIR="%{?buildroot}"
+%makeinstall            make DESTDIR="%{?buildroot:%{buildroot}}" install
 %insserv_prereq         insserv sed
 %fillup_prereq          fillup /usr/bin/mkdir /usr/bin/touch
 # to be removed. deprecated since 11/2019 boo#1152105
