diff --git a/build/parsePrep.c b/build/parsePrep.c
index b704f3e37..393036898 100644
--- a/build/parsePrep.c
+++ b/build/parsePrep.c
@@ -96,9 +96,9 @@ static char *doPatch(rpmSpec spec, uint32_t c, int strip, const char *db,
 
     /* Avoid the extra cost of fork and pipe for uncompressed patches */
     if (notCompressed(fn)) {
-	patchcmd = rpmExpand("%{__patch} ", args, " < ", fn, NULL);
+	patchcmd = rpmExpand("%{__patch} ", args, " < '", fn, "'", NULL);
     } else {
-	patchcmd = rpmExpand("{ %{__rpmuncompress} ", fn, " || echo patch_fail ; } | "
+	patchcmd = rpmExpand("{ %{__rpmuncompress} '", fn, "' || echo patch_fail ; } | "
                              "%{__patch} ", args, NULL);
     }
 
@@ -136,7 +136,7 @@ static char *doUntar(rpmSpec spec, uint32_t c, int quietly)
     }
 
     buf = rpmExpand("%{__rpmuncompress} -x ",
-		    quietly ? "" : "-v ", sp->path, NULL);
+		    quietly ? "" : "-v '", sp->path, "'", NULL);
     rstrcat(&buf,
 	"\nSTATUS=$?\n"
 	"if [ $STATUS -ne 0 ]; then\n"
