--- regina-normal-4.1.2.orig/python/regina-python.in
+++ regina-normal-4.1.2/python/regina-python.in
@@ -163,7 +163,7 @@
 
 if (! $python_cmd) {
     my $which_python;
-    chomp($which_python = `which python`);
+    chomp($which_python = `which python2.3`);
     if (! $which_python) {
         &err("E: A Python interpreter could not be found.\n");
         &err("E: Set \$REGINA_PYTHON to a Python interpreter (such as /usr/bin/python)\n");
