--- a/CMakeLists.txt	2016-12-16 12:20:02.000000000 +0100
+++ b/CMakeLists.txt	2019-08-30 08:56:32.485118980 +0200
@@ -312,7 +312,8 @@
   "http://www.digip.org/jansson/"
   TRUE)
 
-FIND_PACKAGE(Boost 1.55 COMPONENTS iostreams python)
+FIND_PACKAGE(Boost 1.55 COMPONENTS iostreams python27)
+SET(Boost_PYTHON_LIBRARY ${Boost_PYTHON27_LIBRARY})
 INCLUDE_DIRECTORIES(${Boost_INCLUDE_DIR})
 MACRO_LOG_FEATURE(Boost_FOUND
   "Boost (>= 1.55)"
